app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,65 @@
+class CommentsController < ApplicationController
+  before_action :find, except: [:index, :create]
+  before_action :authenticate_user!
+  before_action :is_active?
+  before_action :is_deleted?, except: [:create, :index]
+  before_action :check_access_comment?, only: [:update, :destroy]
+  before_action :check_update_time, only: [:update]
+  before_action :check_delete_time, only: [:destroy]
+
+
+
+  def index
+    render json: CommentBlueprint.render(Comment.all.where :ticket_id => params[:ticket_id], :deleted => false)
+  end
+
+  def show
+    render json: CommentBlueprint.render(@comment)
+  end
+
+  def create
+    comment = Comment.new(new_params)
+    comment.ticket_id = params[:ticket_id] #TODO change permission
+    comment.worker_id = current_user.worker.id
+    if comment.save
+      render json: CommentBlueprint.render(comment), status: :created
+      if comment.message =~ /(@\w{1,20})|(_\w{1,20})/
```

Comment:
move logic into `after_craete` callback on `Ticket` model

Reply:
why ticket not comment model?